File: test_vue.js
const fs = require('fs');
const path = require('path');
const { parse } = require('@vue/compiler-sfc');

function checkDir(dir) {
  const files = fs.readdirSync(dir);
  for (const f of files) {
    const fullPath = path.join(dir, f);
    if (fs.statSync(fullPath).isDirectory()) {
      checkDir(fullPath);
    } else if (fullPath.endsWith('.vue')) {
      const content = fs.readFileSync(fullPath, 'utf-8');
      const res = parse(content);
      if (res.errors.length) {
        console.log('Error in', fullPath);
        console.log(res.errors);
      }
    }
  }
}

checkDir('src');
console.log('Check finished.');

File: test_vue.py
import os, sys
import xml.etree.ElementTree as ET

def check_vue(content):
    import re
    # Just extract <template> wrapper content
    match = re.search(r'<template>(.*)</template>', content, re.DOTALL)
    if not match: return
    tmpl = match.group(1)
    
    # Very crude check: count open vs closed tags for common ones like div, p, span, a, section
    for tag in ['div', 'span', 'p', 'a', 'section', 'button']:
        open_count = len(re.findall(rx_open(tag), tmpl))
        close_count = len(re.findall(rx_close(tag), tmpl))
        if open_count != close_count:
            print(f"Tag <{tag}> mismatch: {open_count} open vs {close_count} close")

def rx_open(tag): return f'<{tag}(\s+[^>]*?)?>'
def rx_close(tag): return f'</{tag}>'

for root, _, files in os.walk('src'):
    for f in files:
        if f.endswith('.vue'):
            path = os.path.join(root, f)
            try:
                content = open(path, encoding='utf-8').read()
                print(f"Checking {f}...")
                check_vue(content)
            except Exception as e:
                print(f"Error reading {f}: {e}")
